lib/auth/auth.js

```diff
@@ -26,120 +30,153 @@ auth.setAuthHandler = handler => {
+    res.params.log = log;
+    if (res.params.version === 2) {
+        return vault.authenticateV2Request(res.params, requestContext, cb);
+    } else if (res.params.version === 4) {
+        return vault.authenticateV4Request(res.params, requestContext, cb,
+                                           awsService);
+    }
+
     log.error('authentication method not found', {
         method: 'Arsenal.auth.doAuth',
     });
     return cb(errors.InternalError);
-};
+}
 
-auth.generateV4Headers =
-    (request, data, accessKey, secretKeyValue, awsService) => {
-        Object.assign(request, { headers: {} });
-        const amzDate = convertUTCtoISO8601(Date.now());
-        // get date without time
-        const scopeDate = amzDate.slice(0, amzDate.indexOf('T'));
-        const signedHeaders = 'host;x-amz-date;x-amz-content-sha256';
-        const region = 'us-east-1';
-        const service = awsService || 'iam';
-        const credentialScope =
-            `${scopeDate}/${region}/${service}/aws4_request`;
-        const timestamp = amzDate;
-        const algorithm = 'AWS4-HMAC-SHA256';
+function generateV4Headers(request, data, accessKey, secretKeyValue,
+                           awsService) {
+    Object.assign(request, { headers: {} });
+    const amzDate = convertUTCtoISO8601(Date.now());
+    // get date without time
+    const scopeDate = amzDate.slice(0, amzDate.indexOf('T'));
+    const signedHeaders = 'host;x-amz-date;x-amz-content-sha256';
+    const region = 'us-east-1';
+    const service = awsService || 'iam';
+    const credentialScope =
+        `${scopeDate}/${region}/${service}/aws4_request`;
+    const timestamp = amzDate;
+    const algorithm = 'AWS4-HMAC-SHA256';
 
-        let payload = '';
-        if (request.method === 'POST') {
-            payload = queryString.stringify(data, null, null, {
-                encodeURIComponent,
-            });
-        }
-        const payloadChecksum = crypto.createHash('sha256').update(payload)
-                                      .digest('hex');
-        const path = request.path;
-        Object.assign(request, { path: '/' });
-        request.setHeader('host', request._headers.host);
-        request.setHeader('x-amz-date', amzDate);
-        request.setHeader('x-amz-content-sha256', payloadChecksum);
-        Object.assign(request.headers, { host: request.getHeader('host') });
-        Object.assign(request.headers, { 'x-amz-date': amzDate });
-        Object.assign(request.headers,
-                      { 'x-amz-content-sha256': payloadChecksum });
+    let payload = '';
+    if (request.method === 'POST') {
+        payload = queryString.stringify(data, null, null, {
+            encodeURIComponent,
+        });
+    }
+    const payloadChecksum = crypto.createHash('sha256').update(payload)
+                                  .digest('hex');
+    const path = request.path;
+    Object.assign(request, { path: '/' });
```

Comment:
This should come out.  I did pr to remove but it appears the rebase did not catch it since the function changed.